cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -74,6 +85,150 @@
+      responder.sendJson(HttpResponseStatus.NOT_FOUND,
+          GSON.toJson("Use PUT to enter repository info first."));
+    } else {
+      JsonObject jObj = DECODE_GSON.fromJson(request.content().toString(StandardCharsets.UTF_8), JsonObject.class);
+      //check that the Json has the necessary fields
+      if (jObj.has("nickname") && jObj.has("url") && jObj.has("default_branch") && jObj.has("path")) {
+        String nickname = jObj.getAsJsonPrimitive("nickname").toString().replace("\"", "");
+        String defaultBranch = jObj.getAsJsonPrimitive("default_branch").toString().replace("\"", "");
+        String path = jObj.getAsJsonPrimitive("path").toString().replace("\"", "");
+
+        String giturl = jObj.getAsJsonPrimitive("url").toString().replace("\"", "");
+        String[] sshUrl = giturl.split(":");
+        int parse = sshUrl[1].indexOf("/");
+        String owner = sshUrl[1].substring(0, parse);
+        String name = sshUrl[1].substring(parse + 1, sshUrl[1].lastIndexOf("."));
+        //connect to GitHub
+        String htmlUrl =  "https://api.github.com/repos/" + owner + "/"
+            + name;
+        URL url = new URL(htmlUrl);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("GET");
+
+        if (jObj.has("username") && jObj.has("password")) {
+          String username = jObj.getAsJsonPrimitive("username").toString().replace("\"", "");
+          String password = jObj.getAsJsonPrimitive("password").toString().replace("\"", "");
+          String authString = "Basic " + Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
+          con.setRequestProperty("Authorization", authString);
+        } else if (jObj.has("authorization")) {
+          String token = jObj.getAsJsonPrimitive("authorization").toString().replace("\"", "");
+          con.setRequestProperty("Authorization", "token " + token);
+        } else {
+          responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Please enter identification.");
+        }
+        //test that the GitHub repository information is valid
+        if (con.getResponseCode() == HttpURLConnection.HTTP_OK) {
+          userRepo = new JsonObject();
+          userRepo.addProperty("nickname", nickname);
+          userRepo.addProperty("url", giturl);
+          userRepo.addProperty("default_branch", defaultBranch);
+          userRepo.addProperty("path", path);
+          userRepo.addProperty("owner", owner);
+          userRepo.addProperty("name", name);
+          responder.sendJson(HttpResponseStatus.OK, GSON.toJson("Repository Info has been updated."));
+        } else {
+          responder.sendJson(HttpResponseStatus.BAD_REQUEST, "This repository does not exist.");
+        }
+      } else {
+        responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Enter fields correctly.");
+      }
+    }
+  }
+
+  /**
+   * Deletes current repository info
+   */
+  @Path("git/repo/delete")
```

Comment:
repos/git/{repo}

similarly DELETE method itself already means it's a deletion.